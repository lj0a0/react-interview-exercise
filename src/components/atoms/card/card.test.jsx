import React from "react";
import Card from './card';
import renderer from "react-test-renderer";


test("Card snapshot ", () => {
  const component = renderer.create(
    <Card
    data={{
        id: 181360063,
        name: "reattempt",
        full_name: "wsmd/reattempt",
        description: "ðŸ¤ž Give your functions another chance",
        html_url: "https://github.com/wsmd/reattempt",
        fork: false /*bool*/,
        forks_count: 11,
        watchers_count: 573,
        stargazers_count: 573,
        updated_at: "2020-02-28T22:22:14Z",
        owner: { login: "wsmd", avatar_url: "https://avatars2.githubusercontent.com/u/2100222?v=4"}
      }}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
