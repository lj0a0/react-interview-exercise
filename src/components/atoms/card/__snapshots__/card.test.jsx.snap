// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card snapshot  1`] = `
<div
  className="col-md-4 mb-4"
>
  <link
    href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css"
    rel="stylesheet"
  />
  <link
    href="https://fonts.googleapis.com/icon?family=Material+Icons"
    rel="stylesheet"
  />
  <div
    className="card shadow-sm h-100"
  >
    <div
      className="card"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "height": 225,
            "width": "100%",
          }
        }
      >
        <img
          alt="..."
          className="card-img-top"
          src="https://avatars2.githubusercontent.com/u/2100222?v=4"
          style={
            Object {
              "height": "80%",
              "margin": "10%",
              "width": "80%",
            }
          }
        />
      </div>
      <div
        className="card-body"
      >
        <div
          style={
            Object {
              "height": 150,
            }
          }
        >
          <h5
            className="card-title"
          >
            reattempt
             from 
            wsmd
          </h5>
          <p
            className="card-text"
          >
            ðŸ¤ž Give your functions another chance
          </p>
        </div>
        <div
          className="container"
          style={
            Object {
              "height": "50%",
            }
          }
        >
          <div
            className="row"
            style={
              Object {
                "margin": "auto",
                "marginBottom": "10px",
              }
            }
          >
            <div
              className="col-sm"
              style={
                Object {
                  "margin": "auto",
                  "marginBottom": "5px",
                  "marginTop": "5px",
                }
              }
            >
              <a
                href="https://github.com/wsmd/reattempt"
                target="_blank"
              >
                <div
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#0367E5",
                      "border": "2px solid #0367E5",
                      "borderRadius": 10,
                      "color": "#FFFFFF",
                      "display": "flex",
                      "height": "100%",
                      "justifyContent": "center",
                    }
                  }
                >
                  View
                </div>
              </a>
            </div>
            <div
              className="col-sm"
              style={
                Object {
                  "margin": "auto",
                  "marginBottom": "5px",
                  "marginTop": "5px",
                }
              }
            >
              <a>
                <div
                  style={
                    Object {
                      "backgroundColor": "#0367E5",
                      "border": "2px solid #0367E5",
                      "borderRadius": 10,
                      "color": "#FFFFFF",
                      "opacity": "0.6",
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <i
                      className="fa fa-eye"
                    />
                     Watch
                  </div>
                  <div
                    style={
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "justifyContent": "center",
                      }
                    }
                  >
                     
                    573
                  </div>
                </div>
              </a>
            </div>
          </div>
          <div
            className="row"
            style={
              Object {
                "margin": "auto",
              }
            }
          >
            <div
              className="col-sm"
              style={
                Object {
                  "margin": "auto",
                  "marginBottom": "5px",
                  "marginTop": "5px",
                }
              }
            >
              <a>
                <div
                  style={
                    Object {
                      "backgroundColor": "#0367E5",
                      "border": "2px solid #0367E5",
                      "borderRadius": 10,
                      "color": "#FFFFFF",
                      "opacity": "0.6",
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <i
                      className="fa fa-star"
                      style={
                        Object {
                          "marginRight": "1px",
                        }
                      }
                    />
                     
                    Star
                  </div>
                  <div
                    style={
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "justifyContent": "center",
                      }
                    }
                  >
                     
                    573
                  </div>
                </div>
              </a>
            </div>
            <div
              className="col-sm"
              style={
                Object {
                  "margin": "auto",
                  "marginBottom": "5px",
                  "marginTop": "5px",
                }
              }
            >
              <a>
                <div
                  style={
                    Object {
                      "backgroundColor": "#0367E5",
                      "border": "2px solid #0367E5",
                      "borderRadius": 10,
                      "color": "#FFFFFF",
                      "opacity": "0.6",
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <i
                      className="fas fa-code-branch"
                      style={
                        Object {
                          "marginRight": "1px",
                        }
                      }
                    />
                     
                    Forks
                  </div>
                  <div
                    style={
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "justifyContent": "center",
                      }
                    }
                  >
                     
                    11
                  </div>
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
